src/mds/Migrator.cc

```diff
@@ -1360,12 +1347,13 @@ void Migrator::export_frozen(CDir *dir, uint64_t tid)
 
   // CDir::_freeze_tree() should have forced it into subtree.
   ceph_assert(dir->get_dir_auth() == mds_authority_t(mds->get_nodeid(), mds->get_nodeid()));
-  // note the bounds.
-  set<CDir*> bounds;
-  mdcache->get_subtree_bounds(dir, bounds);
+  mdcache->get_subtree_bounds(dir, it->second.bounds);
+  for (auto& bound : it->second.bounds)
+   it->second.bound_vec.push_back(bound->dirfrag());
+  std::sort(it->second.bound_vec.begin(), it->second.bound_vec.end());
 
   // generate prep message, log entry.
-  auto prep = make_message<MExportDirPrep>(dir->dirfrag(), it->second.tid);
+  auto prep = make_message<MExportDirPrep>(dir->dirfrag(), it->second.bound_vec, it->second.tid);
```

Comment:
> Follow up question (for my understanding): Is a subtree "bound" a collection of all subtree roots? (where each subtree root could either be a CDir* or a dirfrag_t).

right

Reply:
> The journal entry now has a vector of dirfrags. This used to be just subtree dirfrag and tid earlier. This would increase the size of the journal entry. Would that have other side-effects?

I don't think so